### PR TITLE
feat: log socket exception stack trace periodically

### LINKED ISSUE #14304: Add periodic stack trace printing for socket exception
Add a `RateLimiter` and `RateLimiter::requestAndTrigger()` check to periodically kick the error into the code path that reports the stack trace.  https://github.com/hashgraph/hedera-services/blob/e386407169bf21f81a8d29b2a1cce88b5782d627/platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/network/NetworkUtils.java#L92

```
if (SOCKET_EXCEPTIONS.getMarker().equals(marker) && !stackTraceRateLimiter.requestAndTrigger()) { 
```

### PR DESCRIPTION
**Pull request by anthony-swirldslabs**
**Description**:
Add a RateLimiter and log the stack trace of socket exceptions once a minute.

**Related issue(s)**:

Fixes #14304

**Notes for reviewer**:
Trivial changes to improve logging and diagnostics.

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/network/NetworkUtils.java [hunk 1]
```diff
@@ -90,8 +93,12 @@ public static void handleNetworkException(final Exception e, final Connection co
         // we use a different marker depending on what the root cause is
         final Marker marker = NetworkUtils.determineExceptionMarker(e);
         if (SOCKET_EXCEPTIONS.getMarker().equals(marker)) {
-            final String formattedException = NetworkUtils.formatException(e);
-            logger.warn(marker, "Connection broken: {} {}", description, formattedException);
+            if (socketExceptionRateLimiter.requestAndTrigger()) {
+                logger.warn(marker, "Connection broken: {}", description, e);
```
Comment (kfa-aguda) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/network/NetworkUtils.java:97]: not in a compiler so can't check but I think this should be `Connection broken: {} {}` so the placeholders match the args?

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/network/NetworkUtils.java [hunk 2]
```diff
@@ -90,8 +93,12 @@ public static void handleNetworkException(final Exception e, final Connection co
...
+            } else {
+                final String formattedException = NetworkUtils.formatException(e);
+                logger.warn(marker, "Connection broken: {} {}", description, formattedException);
+            }
         } else {
             logger.error(EXCEPTION.getMarker(), "Connection broken: {}", description, e);
```
Comment (kfa-aguda) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/network/NetworkUtils.java:103]: same thing here though this isn't part of your change. If it is, could you pls fix?

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/network/NetworkUtils.java (hunk 1 above)
Comment (anthony-swirldslabs) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/network/NetworkUtils.java:97]: No. This is a feature of the Logger interface - it accepts one extra argument of type `Throwable`, and when present, it prints its entire stack trace. If there was an explicit `{}` in the formatting string, then the logger would only print the results of the `throwable.toString()` which wouldn't include the stack trace and which would defeat the purpose of this very fix.

This is how you usually want to log all your exceptions, unless they are too noisy like this one, where we have to do it only occasionally using the RateLimiter.

So this code is written by design and there's not a bug w.r.t. the `e`/`{}`.

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/network/NetworkUtils.java (hunk 2 above)
Comment (anthony-swirldslabs) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/network/NetworkUtils.java:103]: Please see above.

**Review by kfa-aguda — APPROVED** [review]

**Pull request merged**